nuxeo-runtime/nuxeo-runtime-test/src/main/java/org/nuxeo/runtime/test/NXRuntimeTestCase.java

```diff
@@ -425,6 +436,67 @@ public RuntimeContext deployTestContrib(String bundle, URL contrib) throws Excep
         return ctx;
     }
 
+    @Override
+    public void deployPartial(String name, Set<TargetExtensions> targetExtensions) throws Exception {
+        // Do not install bundle; we only need the Object to list his components
+        Bundle bundle = new BundleImpl(osgi, lookupBundle(name), null);
+
+        RuntimeContext ctx = new OSGiRuntimeContext(runtime, bundle);
```

Comment:
What I think Julien means is that other methods like `deployTestContrib` return the context to the caller, which may (sometimes) use it to call `destroy` on it to do the undeploy.